pkg/api/customization/catalog/store.go

```diff
@@ -0,0 +1,97 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/rancher/norman/store/proxy"
+	"github.com/rancher/norman/store/transform"
+	"github.com/rancher/norman/types"
+	catUtil "github.com/rancher/rancher/pkg/catalog/utils"
+	hcommon "github.com/rancher/rancher/pkg/controllers/user/helm/common"
+	"github.com/rancher/rancher/pkg/settings"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	managementschema "github.com/rancher/types/apis/management.cattle.io/v3/schema"
+	client "github.com/rancher/types/client/management/v3"
+	"github.com/rancher/types/config"
+)
+
+type templateStore struct {
+	types.Store
+	CatalogTemplateVersionLister v3.CatalogTemplateVersionLister
+}
+
+func GetTemplateStore(ctx context.Context, managementContext *config.ScaledContext) types.Store {
+	ts := templateStore{
+		CatalogTemplateVersionLister: managementContext.Management.CatalogTemplateVersions("").Controller().Lister(),
+	}
+
+	s := &transform.Store{
+		Store: proxy.NewProxyStore(ctx, managementContext.ClientGetter,
+			config.ManagementStorageContext,
+			[]string{"apis"},
+			"management.cattle.io",
+			"v3",
+			"CatalogTemplate",
+			"catalogtemplates"),
+		Transformer: func(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, opt *types.QueryOptions) (map[string]interface{}, error) {
+			data[client.CatalogTemplateFieldVersionLinks] = ts.extractVersionLinks(apiContext, data)
+			return data, nil
+		},
+	}
+
+	ts.Store = s
+
+	return ts
+}
+
+func (t *templateStore) extractVersionLinks(apiContext *types.APIContext, resource map[string]interface{}) map[string]interface{} {
+	schema := apiContext.Schemas.Schema(&managementschema.Version, client.TemplateVersionType)
+	r := map[string]interface{}{}
+	versionMap, ok := resource[client.CatalogTemplateFieldVersions].([]interface{})
+	if ok {
```

Comment:
Is it ok if we return an empty map here?

Reply:
Yes, that's the same as having no versions matching your current rancher version.